platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/cli/BuildActionsFactory.java

```diff
@@ -210,4 +253,28 @@ private long getBuildStartTime() {
     private GradleLauncherMetaData clientMetaData() {
         return new GradleLauncherMetaData();
     }
+
+    private void configDaemonJvm(ParsedCommandLine commandLine, Parameters parameters) {
+        StartParameterInternal startParameters = parameters.getStartParameter();
+        DaemonParameters daemonParameters = parameters.getDaemonParameters();
+        JavaVersion version = jvmVersionDetector.getJavaVersion(daemonParameters.getEffectiveJvm());
+        try {
+            DaemonJvmToolchainSpec jvmToolchainCriteria = buildEnvironmentConfigurationConverter.convertJvmToolchainCriteria(commandLine, parameters.getProperties());
+            if (jvmToolchainCriteria != null) {
+                JvmInstallationMetadata installation = daemonJavaToolchainQueryService.findMatchingToolchain(jvmToolchainCriteria, startParameters);
+                daemonParameters.setJvm(Jvm.forHome(installation.getJavaHome().toFile()));
+                version = installation.getLanguageVersion();
+            }
+        } catch (Exception exception) {
+            if (!isExecutingUpdateDaemonJvmTask(startParameters)) {
+                throw exception;
+            }
+        }
+        daemonParameters.applyDefaultsFor(version);
+    }
+
+    private boolean isExecutingUpdateDaemonJvmTask(StartParameter startParameters) {
+        List<TaskExecutionRequest> taskExecutionRequests = startParameters.getTaskRequests();
```

Comment:
Do we have a test for this behavior? 
As I understand, we ignore a failed search of a requested toolchain if update toolchain task is being executed